tests/rptest/tests/log_segment_ms_test.py

```diff
@@ -0,0 +1,196 @@
+import typing
+
+from ducktape.mark import parametrize, defaults
+from ducktape.utils.util import wait_until
+
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.services.verifiable_consumer import VerifiableConsumer
+from rptest.services.verifiable_producer import VerifiableProducer
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.util import segments_count
+
+# NOTE this value could be read from redpanda
+# for example this should be a reasonable way
+# int(Admin(self.redpanda).get_cluster_config_schema()
+#                       ['properties']['log_segment_ms_min']['example'])
+SERVER_SEGMENT_MS = 60000
+SERVER_HOUSEKEEPING_LOOP = 10
+
+
+def messages_to_generate_segments(num_segments, period):
+    return int(period / 1000 * (num_segments) + SERVER_HOUSEKEEPING_LOOP + 1)
```

Comment:
This function doesn't make sense to me: if it's returning a message count, why would it need to produce SERVER_HOUSEKEEPING_LOOP +1 messages?

e.g. for 1 segment with segment.ms=60000, this function is returning `71`.  Why does 71 messages result in one segment?

Reply:
The throughput on the verifiable producer is set to 1 message per second, so this functiona basically returns the number of seconds until we'll have generated `num_segments` via rolling.